From 319d6d68a398d326ef95e39f494da7c514facf81 Mon Sep 17 00:00:00 2001
From: Sobhan Mohammadpour <sobhanmohammadpour1@yahoo.fr>
Date: Fri, 8 Mar 2013 08:25:58 +0330
Subject: [PATCH] pygobject-3.7.90-make check

originaly by [Alexandre Rostovtsev <tetromino@gmail.com>
don't build tests unless it's needed

---
 tests/Makefile.am | 12 ++++++------
 tests/runtests.py |  3 +++
 2 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index 287542d..2cc0da9 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -1,5 +1,5 @@
 CLEANFILES =
-noinst_LTLIBRARIES = libgimarshallingtests.la
+check_LTLIBRARIES = libgimarshallingtests.la
 test_typelibs = GIMarshallingTests-1.0.typelib
 
 nodist_libgimarshallingtests_la_SOURCES = $(GI_DATADIR)/tests/gimarshallingtests.c $(GI_DATADIR)/tests/gimarshallingtests.h
@@ -23,7 +23,7 @@ GIMarshallingTests-1.0.typelib: GIMarshallingTests-1.0.gir Makefile
 
 # regress.c needs cairo
 if ENABLE_CAIRO
-noinst_LTLIBRARIES += libregress.la
+check_LTLIBRARIES += libregress.la
 test_typelibs += Regress-1.0.typelib
 nodist_libregress_la_SOURCES = $(GI_DATADIR)/tests/regress.c $(GI_DATADIR)/tests/regress.h
 libregress_la_CFLAGS = $(GIO_CFLAGS) $(CAIRO_CFLAGS)
@@ -52,7 +52,7 @@ gschemas.compiled: org.gnome.test.gschema.xml
 
 CLEANFILES += Regress-1.0.gir Regress-1.0.typelib GIMarshallingTests-1.0.gir GIMarshallingTests-1.0.typelib gschemas.compiled
 
-noinst_LTLIBRARIES += testhelper.la
+check_LTLIBRARIES += testhelper.la
 
 testhelper_la_CFLAGS = -I$(top_srcdir)/gi/_gobject -I$(top_srcdir)/gi/_glib $(PYTHON_INCLUDES) $(GLIB_CFLAGS)
 testhelper_la_LDFLAGS = -module -avoid-version
@@ -72,7 +72,7 @@ testhelper.la: $(testhelper_la_OBJECTS) $(testhelper_la_DEPENDENCIES)
 	test -L $@ || $(LN_S) .libs/$@ $@
 
 
-all: $(LTLIBRARIES:.la=.so)
+all: $(check_LTLIBRARIES:.la=.so)
 
 EXTRA_DIST = \
 	compathelper.py \
@@ -116,7 +116,7 @@ EXTRA_DIST = \
 	$(NULL)
 
 clean-local:
-	rm -f $(LTLIBRARIES:.la=.so) file.txt~
+	rm -f $(check_LTLIBRARIES:.la=.so) file.txt~
 
 DBUS_LAUNCH=$(shell which dbus-launch)
 RUN_TESTS_ENV_VARS= \
@@ -130,7 +130,7 @@ RUN_TESTS_ENV_VARS= \
 
 # pygtkcompat tests need to be run in a separate process as they
 # clobber global name space
-check-local: $(LTLIBRARIES:.la=.so) $(test_typelibs) gschemas.compiled
+check-local: $(check_LTLIBRARIES:.la=.so) $(test_typelibs) gschemas.compiled
 	@echo "  CHECK  Pyflakes"
 	@if type pyflakes >/dev/null 2>&1; then pyflakes $(top_srcdir); else echo "skipped, pyflakes not installed"; fi
 	@if test -z "$$SKIP_PEP8"; then \
diff --git a/tests/runtests.py b/tests/runtests.py
index d3a4f47..5a53d3f 100755
--- a/tests/runtests.py
+++ b/tests/runtests.py
@@ -54,6 +54,9 @@ if sys.version_info[:2] == (2, 6):
 if sys.version_info[:2] == (2, 7):
     unittest.TestCase.assertRaisesRegex = unittest.TestCase.assertRaisesRegexp
 
+# Some tests fail with translated messages.
+os.environ["LC_ALL"] = "C"
+
 if '--help' in sys.argv:
     print("Usage: ./runtests.py <testfiles>")
     sys.exit(0)
-- 
1.8.1.2

